src/providers/MeetingProvider/MeetingManager.ts

```diff
@@ -326,8 +326,8 @@ export class MeetingManager implements AudioVideoObserver {
     // This results into missing few meeting events triggered with audioVideoDidStop such as meetingEnded, meetingFailed and meetingStartFailed.
     // We may also loose audioInputUnselected and videoInputUnselected events as we choose null devices when an attendee leaves.
     // When a new AudioVideoFacade object is created remove and re-add the eventDidReceive observer which wont leak.
-    this.audioVideo.removeObserver(this.eventDidReceiveRef);
-    this.audioVideo.addObserver(this.eventDidReceiveRef);
+    this.eventController.removeObserver(this.eventDidReceiveRef);
```

Comment:
This should be unnecessary now but I'll check

Reply:
Yes, we can remove this now. Have to check a bit more on where to clean-up if needed so that we do not keep the added observers around when not needed.

Reply:
Changed where these are called, will try to remove in leave and add in initializeMeetingSession